Framework/Blockhub.Core/Services/Wallet/SearchForAccountsByBalanceWalletCreate.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Blockhub.Services;
+using Blockhub.Wallet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Blockhub.Wallet
+{
+    public class SearchForAccountsByBalanceWalletCreate<T> : IWalletCreate<T> where T : IBlockchainType
+    {
+        private IWalletCreate<T> WalletCreator { get; }
+        private IAccountCreate<T> WalletManager { get; }
+        private IBalanceRead<T> BalanceReader { get; }
+        private int SearchCount { get; }
+
+        public SearchForAccountsByBalanceWalletCreate(IWalletCreate<T> walletCreator, IAccountCreate<T> walletManager, IBalanceRead<T> balanceReader) :
+            this(walletCreator, walletManager, balanceReader, 100)
+        { }
+
+        public SearchForAccountsByBalanceWalletCreate(IWalletCreate<T> walletCreator, IAccountCreate<T> walletManager, IBalanceRead<T> balanceReader, int searchCount)
+        {
+            WalletCreator = walletCreator ?? throw new ArgumentNullException(nameof(walletCreator));
+            WalletManager = walletManager ?? throw new ArgumentNullException(nameof(walletManager));
+            BalanceReader = balanceReader ?? throw new ArgumentNullException(nameof(balanceReader));
+
+            if (searchCount < 1) throw new ArgumentOutOfRangeException(nameof(searchCount), "Search count must be at least one.");
+            SearchCount = searchCount;
+        }
+
+        public async Task<Wallet<T>> CreateWallet(string secret, string name, string password)
+        {
+            return await SearchAndFillInAccounts(await WalletCreator.CreateWallet(secret, name, password));
+        }
+
+        public async Task<Wallet<T>> CreateWallet(string name, string password)
+        {
+            return await SearchAndFillInAccounts(await WalletCreator.CreateWallet(name, password));
+        }
+
+        private async Task<Wallet<T>> SearchAndFillInAccounts(Wallet<T> wallet)
```

Comment:
Seems a little strange that this class takes a wallet and returns the same wallet, modifying the state within. Do you think it would be better if this was `SearchForAccounts` and returned the list of accounts? The user can then add if they want, or maybe they just want to check them.

Reply:
I will change this for simplicity. The other way is better when we want to add functionality without changing code in the UI. It auto-fills in the accounts and follows the open/closed principle 

[Uncle Bob](https://8thlight.com/blog/uncle-bob/2014/05/12/TheOpenClosedPrinciple.html)

[Example](http://joelabrahamsson.com/a-simple-example-of-the-openclosed-principle/)

I used decoration to perform the open/closed principle here.